src/models/activity.rs

```diff
@@ -216,7 +218,7 @@ impl Activity {
         goal: &Goal,
         calendar: &Calendar,
     ) -> Vec<Activity> {
-        if goal.children.is_some() || goal.filters.as_ref().is_none() {
+        if goal.filters.as_ref().is_none() {
```

Comment:
This is good. I also removed goal.children.is_some() in the filler goal test case. Therefore ok.